rflx/generator/common.py

```diff
@@ -116,17 +118,51 @@ def byte_aggregate(aggregate: Aggregate) -> Aggregate:
                         ),
                         aggregate,
                     )
-                return Call(
-                    "Equal", [Variable("Ctx"), Variable(Field(field.name).affixed_name), aggregate]
-                )
+                return Call("Equal", [Variable("Ctx"), Variable(field.affixed_name), aggregate])
+
+        literals = [
+            l
+            for t in message.types.values()
+            if isinstance(t, Enumeration)
+            for l in t.literals.keys()
+        ]
+
+        def field_value(field: Field) -> Expr:
+            if public:
+                return Call(f"Get_{field.name}", [Variable("Ctx")])
+            return Selected(
+                Indexed(
+                    Variable("Ctx.Cursors" if not embedded else "Cursors"),
+                    Variable(field.affixed_name),
+                ),
+                f"Value.{field.name}_Value",
+            )
+
+        if isinstance(expression, Relation):
+            for left, right in [
+                (expression.left, expression.right),
+                (expression.right, expression.left),
+            ]:
+                if (
+                    isinstance(left, Variable)
+                    and Field(left.name) in message.fields
+                    and (
+                        isinstance(right, Number)
+                        or (isinstance(right, Variable) and right.name in literals)
+                    )
+                ):
+                    return expression.__class__(field_value(Field(left.name)), right)
```

Comment:
I'm not sure I understand this part correctly, but if in the `for` loop we have the situation that `left == expression.right` and `right == expression.left` wouldn't the return statement actually mean

```Python
   return expression.__class__(field_value(Field(expression.right.name)), expression.left)
```

and thus flipping operands, which would be an issue for non-commutative relations like `Less`?

Reply:
You are right. I have tried too hard to avoid code duplication. Fixed by f70e4f9.